Resources/doc/reference/form_types.rst

```diff
@@ -370,6 +370,9 @@ According the choice made only associated fields are displayed. The others field
 map
   Associative array. Describes the fields that are displayed for each choice.
 
+**TIP**: To mask a subfield replace `.` by `__`.
```

Comment:
If there is a third entry in my form like this

```
 $formMapper->add('type', ...)
      ->add('characteristic.puissance', ...)
      ->add('characteristic__puissance', 'text')
```

I imagine symfony escaped the `__` to avoid conflict, so the fix should manage this too, and if someone as replaced its `.` per `__` his code will be broken.
I'm ok to make a pr but I want to be sure there is problem with that.